### PR TITLE
Add support for github actions variables

### LINKED ISSUE #1534: [FEAT]: Environment Variables for Github Actions
### Describe the need

Github created the possibility to set environment variables for github actions. This feature is already reflected in the newest REST API version of github: 2022-11-28.
Documentation example [here](https://docs.github.com/en/rest/actions/variables?apiVersion=2022-11-28#create-an-environment-variable).
It seems like, the environment variables are pretty similar to the already here implemented environment secrets.

### SDK Version

_No response_

### API Version

2022-11-28

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by tjcorr**
Resolves #1534 

----

### After the change?

Adds 3 new resources to support GitHub Actions variables at the repo, environment, and organization levels:
- `github_actions_variable`
- `github_actions_environment_variable`
- `github_actions_organization_variable`

In addition this also adds the corresponding data sources:
- `github_actions_variables`
- `github_actions_environment_variables`
- `github_actions_organization_variables`
 
As well as the missing data source `github_actions_environment_secrets`

### Other information

This required an update to a new version of the github.com/google/go-github/v50 module (now v50.1.0) to support the new APIs.

----

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

### Pull request type

 
 

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Feature/model/API additions: `Type: Feature`
- Updates to docs or samples: `Type: Documentation`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

**Comment by tjcorr** [comment]
I wasn't sure if there was a process to auto-generate the docs for the new resources/data sources. If not just let me know and I can try and write some.

**Comment by Eladrian05** [comment]
Futuro

**Review by Eladrian05 — APPROVED** [review]

**Comment by ilmax** [comment]
> I wasn't sure if there was a process to auto-generate the docs for the new resources/data sources. If not just let me know and I can try and write some.

Hello @tjcorr , I asked the same a while ago [here](https://github.com/integrations/terraform-provider-github/pull/1530#issuecomment-1435423729)

**Comment by tjcorr** [comment]
I added in support for importing existing variables as well as the new docs pages. Should be ready for a review at this point.

File: website/docs/d/actions_environment_secrets.html.markdown [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
```
Comment (kfcampbell) [inline website/docs/d/actions_environment_secrets.html.markdown:1]: @tjcorr do you mind linking to these excellent new docs in the `github.erb` file?

**Review by kfcampbell — COMMENTED** [review]
This looks really good, thank you @tjcorr! If you don't want to/don't have time to link the docs, let me know and I can take care of it before getting this in.

**Comment by yacinehmito** [comment]
Thank you very much @tjcorr! Can't wait for the release.

**Review by kfcampbell — APPROVED** [review]
@tjcorr You're the best, thank you!

**Pull request merged**